server.py

```diff
@@ -6,29 +6,39 @@
 
 app = Bottle()
 
-with open('./files/expected_outputs/expected_output.txt', 'r') as fl:
-    expected = fl.read()
+question_dir = 'files/questions'
+questions = {}
+for qno in os.listdir(question_dir):
+    questions[qno] = {}
+    with open(os.path.join(question_dir, qno, 'expected.txt'), 'r') as fl:
+        questions[qno]['expected'] = fl.read()
+    with open(os.path.join(question_dir, qno, 'statement.txt'), 'r') as fl:
+        questions[qno]['statement'] = fl.read()
+
 
 @app.get('/question/<number>')
 def question1(number):
-    return static_file('index.html', root=dir_path)
-
-@app.get('/question/<number>/download/')
-def download(number):
-    input_file_name = 'inputs'+number+'.txt'
-    return static_file(input_file_name, root=dir_path+'/'+'files/questions/')
-
-@app.route('/upload', method = 'POST')
-def file_upload():
+    global questions
+    statement = questions[number]['statement']
+    return template('question.html', qno=number, statement=statement)
+
+@app.get('/question/<path:path>')
```

Comment:
What does `<path:path>` mean?

Reply:
Bottle has [dynamic routes](https://bottlepy.org/docs/dev/tutorial.html#dynamic-routes) which allow you to call the same function for different urls. That is what we are using here. With this, `/question/1/input.txt`, `/question/1/statement.txt`, `/question/2/input.txt` (in fact anything starting with `/question` and having something after that is handled by this function.

The rest of the URL (`/1/input.txt`, `/1/statement.txt` etc) is passed on to the function in the variable called path. We use that variable for static file serving.